usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/crossvalidation/StopTimeBasedCrossValidator.java

```diff
@@ -35,38 +38,50 @@
     private final ValidationResultRepository resultRepo;
     private final GtfsDataRepository dataRepo;
     private final Logger logger;
+    private final TimeUtils timeUtils;
     private final ValidateShapeIdReferenceInStopTime validateShapeIdReferenceInStopTime;
     private final ValidateStopTimeTripId validateStopTimeTripId;
+    private final ValidateStopTimeTimeCombination validateStopTimeTimeCombination;
 
     public StopTimeBasedCrossValidator(final GtfsDataRepository dataRepo,
                                        final ValidationResultRepository resultRepo,
                                        final Logger logger,
+                                       final TimeUtils timeUtils,
                                        final ValidateShapeIdReferenceInStopTime validateShapeIdReferenceInStopTime,
-                                       final ValidateStopTimeTripId validateStopTimeTripId
+                                       final ValidateStopTimeTripId validateStopTimeTripId,
+                                       final ValidateStopTimeTimeCombination validateStopTimeTimeCombination
     ) {
         this.resultRepo = resultRepo;
         this.dataRepo = dataRepo;
         this.logger = logger;
+        this.timeUtils = timeUtils;
         this.validateShapeIdReferenceInStopTime = validateShapeIdReferenceInStopTime;
         this.validateStopTimeTripId = validateStopTimeTripId;
+        this.validateStopTimeTimeCombination = validateStopTimeTimeCombination;
     }
 
     /**
      * Executes cross validation rules based on file `stop_times.txt`
      */
     public void execute() {
-        logger.info("Validating rules :'E034 - `shape_id` not found");
+        logger.info("Validating rules: 'E047 - Bad combination of stoptime arrival and departure times`");
+        logger.info("                  'E034 - `shape_id` not found");
         logger.info("                  'E037 - `trip_id` not found");
 
-        dataRepo.getStopTimeAll().values().forEach(stopTimeCollection ->
-                stopTimeCollection.values().forEach(stopTime -> {
-                    final Trip trip = dataRepo.getTripById(stopTime.getTripId());
-                    final Map<Integer, ShapePoint> shape = dataRepo.getShapeById(trip == null ? null : trip.getShapeId());
-                    // E034
-                    validateShapeIdReferenceInStopTime.execute(resultRepo, stopTime, shape, trip);
-                    // E037
-                    validateStopTimeTripId.execute(resultRepo, stopTime, dataRepo.getTripAll());
-                })
+        StopTime previousStopTime = null;
+
+        dataRepo.getStopTimeAll().values().forEach(stopTimeCollection -> {
+                    // E046
+                    validateStopTimeTimeCombination.execute(resultRepo, stopTimeCollection, timeUtils);
```

Comment:
I wouldn't really call this "cross validating", since it's fully enclosed within stop_times.txt. Maybe the `StopTimeBasedCrossValidator` class should be renamed? When I hear "cross validating" I picture comparing one file against another.

Reply:
That's true, and it's actually kind of confusing by the end. I suggest renaming it as `StopTimeValidator`, we could add comments to specify when a use case executes cross validation for more legibility.

Reply:
I like `StopTimeValidator` if these are all related to stop_times.txt - that's very clear.